src/styles/blocks/calendar-loops.scss

```diff
@@ -0,0 +1,23 @@
+@each $day, $margin in $days {
+  .calendar--#{$day} > .calendar__day:first-child {
+    margin-left: ($block-size + 1) * $margin;
+  }
+}
+
+@for $i from 1 through 31 {
+  .calendar > .calendar__day:nth-child(#{$i})::before {
+    content: "#{$i}";
+  }
+}
+
+@for $i from 28 through 31 {
+  .calendar--#{$i} > .calendar__day:nth-child(n + #{$i}) {
+    display: none;
+  }
+}
+
+@for $i from 28 through 31 {
+  .calendar--#{$i} > .calendar__day:nth-child(-n + #{$i}) {
+    display: flex;
+  }
+}

```

Comment:
I think it's better to write your loops inside blocks which they belong to. The reason being that you can use & (a sass shorthand for a parent element) and it makes more sense.